### PR TITLE
Information for new contributors

### PR DESCRIPTION
**Pull request by nrc**

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -45,11 +56,150 @@ that toml file located in `./tests/config/` for its configuration. Including
...
 
 Here are some [good starting issues](https://github.com/nrc/rustfmt/issues?q=is%3Aopen+is%3Aissue+label%3Aeasy).
-Note than some of those issues tagged 'easy' are not that easy and might be better
-second issues, rather than good first issues to fix.
 
 If you've found areas which need polish and don't have issues, please submit a
 PR, don't feel there needs to be an issue.
+
+
+### guidelines
```
Comment (marcusklaas) [inline Contributing.md]: Missing capital?

File: Contributing.md [hunk 2]
```diff
@@ -45,11 +56,150 @@ that toml file located in `./tests/config/` for its configuration. Including
...
+
+When formatting the `bar` call we will format the arguments in order, after the
+first one we know we are working on multiple lines (imagine it is longer than
+written). So, when we come to the second argument, the indent we pass to
+`rewrite` is 12, which puts us under the first argument. The current block
+indent (stored in the context) is 8. The former is used for visual indenting
+(when objects are vertically aligned with some marker), the latter is used for
+block indenting (when objects are tabbed in from the lhs). The width available
+for `baz()` will be the maximum width, minus the space used for indenting, minus
+the space used for the `);`. (Note actual argument formatting does not quite
```
Comment (marcusklaas) [inline Contributing.md]: Note *that*...

**Comment by marcusklaas** [comment]
:100:

Very nice!

r+ with the two typos fixed

**Comment by marcusklaas** [comment]
Could it be useful to link to the [ast docs](http://manishearth.github.io/rust-internals-docs/syntax/ast/index.html) somewhere? It has been most helpful to me in the past.

**Comment by nrc** [comment]
typos corrected and link added

**Pull request merged**